editor/src/components/canvas/plugins/style-plugins.ts

```diff
@@ -1,33 +1,207 @@
+    styleInfo: StyleInfo | null,
+    updatedProperties: StylePropsUpdatedDuringInteraction,
+  ) => ValueAtPath[]
+}
+
+const genericPropPatcher =
+  (zeroValue: string) =>
+  (
+    prop: string,
+    styleInfo: StyleInfo | null,
+    updatedProperties: StylePropsUpdatedDuringInteraction,
+  ) => {
+    const propIsSetOnElement =
+      (styleInfo as Record<string, { type: string }> | null)?.[prop]?.type === 'property' // TODO type
+    const propIsSetFromCommand = updatedProperties.propertiesUpdated.includes(prop)
+    if (!propIsSetOnElement || propIsSetFromCommand) {
+      return []
+    }
+    return [makeZeroProp(prop, zeroValue)]
+  }
+
+const patchers: PropPatcher[] = [
+  { matches: (p) => p === 'gap', patch: genericPropPatcher('0px') },
+  {
+    matches: (p) => p === 'padding',
+    patch: (_, styleInfo, updatedProperties) => {
+      const propIsSetOnElement = styleInfo?.padding != null
+
+      if (!propIsSetOnElement) {
+        return []
+      }
+
+      return PaddingLonghands.filter((p) => !updatedProperties.propertiesUpdated.includes(p)).map(
+        (p) => makeZeroProp(p),
+      )
+    },
+  },
+  { matches: (p) => p.startsWith('padding'), patch: genericPropPatcher('0px') },
+]
+
+function getPropertiesToZero(
+  styleInfo: StyleInfo | null,
+  updatedProperties: StylePropsUpdatedDuringInteraction,
+): ValueAtPath[] {
+  return updatedProperties.propertiesDeleted.flatMap((prop): ValueAtPath[] => {
+    const match = patchers.find((p) => p.matches(prop))
+    if (match == null) {
+      return []
+    }
+    return match.patch(prop, styleInfo, updatedProperties)
+  })
+}
+
+export function patchRemovedProperties(editorState: EditorState): EditorState {
+  const styleInfoReader = getActivePlugin(editorState).styleInfoFactory({
+    projectContents: editorState.projectContents,
+    metadata: editorState.jsxMetadata,
+    elementPathTree: editorState.elementPathTree,
+  })
+
+  return Object.entries(UpdatedPropertiesGlobal.current).reduce(
+    (acc, [elementPathString, updatedProperties]) => {
+      const elementPath = EP.fromString(elementPathString)
+      const styleInfo = styleInfoReader(elementPath)
+      const propsToZero = getPropertiesToZero(styleInfo, updatedProperties)
+      if (propsToZero.length === 0) {
+        return acc
+      } else {
+        return applyValuesAtPath(acc, elementPath, propsToZero).editorStateWithChanges
+      }
+    },
+    editorState,
+  )
+}
+
+export function runStyleUpdateForStrategy(
+  commandLifecycle: InteractionLifecycle,
+  editorState: EditorState,
+  elementPath: ElementPath,
+  updates: StyleUpdate[],
+) {
+  switch (commandLifecycle) {
+    case 'mid-interaction':
+      return runStyleUpdateMidInteraction(editorState, elementPath, updates)
+    case 'end-interaction':
+      return getActivePlugin(editorState).updateStyles(editorState, elementPath, updates)
+    default:
+      assertNever(commandLifecycle)
+  }
+}
```

Comment:
writing the inline styles instead of the whatever the active plugins needs during an interaction